app/scripts/modules/ecs/src/serverGroup/configure/wizard/capacityProvider/CapacityProvider.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import { module } from 'angular';
+import { react2angular } from 'react2angular';
+import {IEcsCapacityProviderStrategy, IEcsServerGroupCommand} from '../../serverGroupConfiguration.service';
+import { HelpField, withErrorBoundary } from  '@spinnaker/core';
+
+export interface ICapacityProviderProps {
+  command: IEcsServerGroupCommand;
+  notifyAngular: (key: string, value: any) => void;
+  configureCommand: (query: string) => PromiseLike<void>;
+}
+
+interface ICapacityProviderState {
+  capacityProviderStrategy: IEcsCapacityProviderStrategy[],
+}
+
+class CapacityProvider extends React.Component<ICapacityProviderProps, ICapacityProviderState>{
+  constructor(props: ICapacityProviderProps) {
+    super(props);
+    const cmd = this.props.command;
+
+    this.state = {
+      capacityProviderStrategy: cmd.capacityProviderStrategy,
+    };
+  }
+
+  private pushCapacityProviderStrategy = () => {
+    const capacityProviderStrategy = this.state.capacityProviderStrategy;
+    capacityProviderStrategy.push({ capacityProvider: '', base: null, weight: null});
+    this.setState({ capacityProviderStrategy : capacityProviderStrategy });
+  };
+
+  private removeCapacityProviderStrategy = (index: number) => {
+    const capacityProviderStrategy = this.state.capacityProviderStrategy;
+    capacityProviderStrategy.splice(index, 1);
+    this.setState({capacityProviderStrategy : capacityProviderStrategy });
+  }
+
+  private updateCapacityProviderName = (index: number, targetCapacityProviderName: string) => {
+    const currentCapacityProviderStartegy = this.state.capacityProviderStrategy;
+    const targetCapacityProviderStrategy = currentCapacityProviderStartegy[index];
+    targetCapacityProviderStrategy.capacityProvider = targetCapacityProviderName;
```

Comment:
did you test adding a new capacity provider strat item with this UI and then deploying it? Any mutation to `capacityProviderStrategy` needs to be saved in the parent (angular) command as well w/ `notifyAngular()` (see [here](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/ecs/src/serverGroup/configure/wizard/taskDefinition/TaskDefinition.tsx#L145)) or else you're only updating the state of the _component_ data, but not the top-level command object (which gets persisted to the pipeline)

Reply:
ummm, that was a great catch. As a part of testing I added CP strat item to the pipeline via UI and later changed the code without  notifyAngular() and it was taking CP details from the pipeline. But now I understood the flow between angular and react so I will make the changes accordingly. Thanks for this catch @allisaurus 👍